site/website/versioned_docs/version-v1.0.0/3_manage_nifi/1_manage_clusters/1_deploy_cluster/3_expose_cluster/2_istio_service_mesh.md

```diff
@@ -3,3 +3,135 @@ id: 2_istio_service_mesh
 title: Istio service mesh
 sidebar_label: Istio service mesh
 ---
+
+The purpose of this section is to explain how expose your NiFi cluster using Istio on kubernetes.
+
+## Istio configuration for HTTP
+
+To access to the NiFi cluster from the external world, it is needed to configure a gateway and a virtual service on Istio.
+
+The following example show ho to define a gateway that will be able to intercept all requests for a specific domain host on http port 80.
+
+```yaml
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: nifi-gateway
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - nifi.my-domain.com
+```
+
+In combination, we need to define also a virtual service that redirect all requests itercepted by the gateway to a specific service. 
+
+```yaml
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: nifi-vs
+spec:
+  gateways:
+  - nifi-gateway
+  hosts:
+  - nifi.my-domain.com
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: nifi
+        port:
+          number: 8080
+```
+
+## Istio configuration for HTTPS
+
+In case you are deploying a cluster and you want enable the HTTPS protocol to manage user authentication, the configuration is more complex. To understand the reason of this, it is important to explain that in this scenario the HTTPS protocol with all certificates is managed directly by NiFi. This means that all requests passes through all Istio services in an encrypted way, so Istio can't manage a sticky session.
+To solve this issue, the tricky is so limit the HTTPS session till the gateway, then decrypt all requests in HTTP, manage the sticky session and then encrypt again in HTTPS before reach the NiFi node.
+Istio allows to do this using a destination rule combined with the virtual service. In the next example we define a gateway that accepts HTTPS traffic and transform it in HTTP.
```

Comment:
gateway -> `Gateway`
virtual service -> `VirtualService`
In the next example we define -> In the next example, we define